app/models/thread.php

```diff
@@ -75,4 +75,63 @@ public static function countAll()
         $db = DB::conn();
         return (int) $db->value("SELECT COUNT(*) FROM thread");
     }
+
+    public function edit($thread_id)
+    {
+        if (!$this->title) {
+            throw new RecordNotFoundException('Record Not Found');
+        }
+
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->query('UPDATE thread SET title = ?, created = NOW()
+                    WHERE id = ?', array($this->title, $thread_id));
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public static function getThread($thread_id)
+    {
+        $user = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread WHERE id = ?", array($thread_id));
+
+        foreach ($rows as $row) {
+            $user[] = new self($row);
+        }
+        return $user;
+    }
+
+    public static function delete($thread_id)
+    {
+        $db = DB::conn();
+        try {
+            $db->begin();
+            $db->query('DELETE FROM thread WHERE id = ?', array($thread_id));
+            $db->query('DELETE FROM comment WHERE thread_id = ?', array($thread_id));
```

Comment:
i think delete comment query is not the right place here since this is in thread class.
you can move it to comment class [like Comment::deleteByThreadId($commend_id)]
you can also make a service class

```php
class BoardService
{
    public static function deleteByThreadId($thread_id)
    {
        Thread::deleteById($thread_id);
        Comment::deleteByThreadId($thread_id);
    }
}
```